### ISSUE TITLE
[SYCL][PI][HIP]

### ISSUE THREAD

**Issue by sommerlukas** [intel/llvm#6189]
Hi, 
**Describe the bug**

I today encountered a compilation error when trying to build the HIP PI plugin on with HIP on a CUDA machine. 

**To Reproduce**

The error is caused by an illegal `reinterpret_cast` in the implementation of `hip_piextMemGetNativeHandle` introduced in [this commit](https://github.com/intel/llvm/commit/e06d1b5fc71998ccea49dce1d8ed4c295c4c4ef9#diff-b88f648055cd14bce67433ad94075c14ff434a79d971857da7cf1a5daf596dd4R2104). 

The underlying problem is that this code attempts to cast a `native_type == hipDeviceptr_t == CUdeviceptr == unsigned long long` to a `pi_native_handle == unsigned long`. 

Running
```
python ../buildbot/configure.py -t RelWithDebInfo --shared-libs --use-lld --hip --hip-platform=Nvidia -o ./
ninja sycl-toolchain
```
results in 
```
.../llvm/sycl/plugins/hip/pi_hip.cpp: In function ‘pi_result hip_piextMemGetNativeHandle(pi_mem, pi_native_handle*)’:
.../llvm/sycl/plugins/hip/pi_hip.cpp:2126:69: error: invalid cast from type ‘_pi_mem::mem_::buffer_mem_::native_type {aka long long unsigned int}’ to type ‘pi_native_handle {aka long unsigned int}’
       reinterpret_cast<pi_native_handle>(mem->mem_.buffer_mem_.get());
```

**Environment (please complete the following information):**

- OS: Linux
- Target device and vendor: Nvidia RTX 3060
- DPC++ version: commit 53fc8e453c60dd3ed1fdaa81e65d5f5d58a05848, "Date:   Tue May 24 23:03:16 2022 -0700"
- Dependencies version: CUDA 11.6, HIP 4.4
 
**Additional context**
Replacing the `reinterpret_cast` by `static_cast` avoids the compilation error, but might be unsafe due to narrowing.

**Comment by zjin-lcf** [comment]
Do you see the problem when you build the CUDA PI on a CUDA machine and build the HIP PI on a HIP machine ?

**Comment by sommerlukas** [comment]
The problem occurs when building the HIP PI on a CUDA machine.

I currently do not have an AMD machine available to test the HIP PI on a HIP machine and I haven't tested the CUDA PI.

**Comment by zjin-lcf** [comment]
pi_cuda.cpp has _PI_CL(piextMemGetNativeHandle, cuda_piextMemGetNativeHandle)

**Comment by sommerlukas** [comment]
The implementation in pi_cuda.cpp uses a `static_cast`, I've opened a pull request using that solution for the HIP plugin: #6207

**Comment by sommerlukas** [comment]
Closed by #6207
